### PR TITLE
Add support for 'su' subshell. Closes #3

### LINKED ISSUE #3: log.sh returns errors when run within "sudo su <someuser> -" style sub-shell
Steps to reproduce:

> \> sudo su SOMEUSER -
> \> source log.sh
> \> LSWARNING test
-bash: /dev/stdout: Permission denied

This seems to be directly related to this issue:
http://unix.stackexchange.com/questions/38538/bash-dev-stderr-permission-denied

log.sh is attempting to write to "/dev/stdout", and this fails if the current user isn't the user who owns whatever pseudoterminal /dev/stdout is symlinked to.

Per Mikel or Keith on StackExchange:

> /dev/stdout is a symlink to /proc/self/fd/1, which is a symlink to, for example, /dev/pts/1. /dev/pts/1, which is a pseudoterminal, is owned by, and writable by, username; that ownership was granted when username logged in. When you sudo su username2, the ownership of /dev/pts/1 doesn't change, and username2 doesn't have write permission.

Line 25 of log.sh is:
> LS_OUTPUT=${LS_OUTPUT:-/dev/stdout}

I tried changing that to:
> LS_OUTPUT=${LS_OUTPUT:-&1}

But that didn't work.

### PR DESCRIPTION
**Pull request by gregorydulin**

### CODE REVIEW

File: log.sh [hunk 1]
```diff
@@ -76,9 +76,16 @@ LSLOG () {
   # if no message was passed, read it from STDIN
   local _MSG
   [[ $# -ne 0 ]] && _MSG="$@" || _MSG="$(cat)"
-  echo -ne "$_LS_LEVEL_BEGIN$OUTPUT " >> "$LS_OUTPUT"
-  echo -n  "$_MSG"                    >> "$LS_OUTPUT"
-  echo -e "$_LS_LEVEL_END" >> "$LS_OUTPUT"
+  if [[ "$LS_OUTPUT" = "/dev/stdout" ]]
+  then
```
Comment (livibetter) [inline log.sh]: I missed this one, I prefer `then` on the if line:

```
if ... ; then
```

Please do a commit with `--amend` after you change it (don't forget to add first or commit with `-a`), then push with `-f` (force).

Comment (gregorydulin) [inline log.sh]: I'm using the github.com web interface, and I didn't see any --amend style options.

This guide says you should be able to flatten all commits in my branch into a single commit on merge, though:
https://github.com/blog/2141-squash-your-commits

If you're using the CLI, though, I don't have a clue how you might do that.  If that's the case, I'll just make another fork/edit/PR.

Comment (livibetter) [inline log.sh]: I have to ask you to do another PR, sorry.
